src/backend/distributed/connection/connection_management.c

```diff
@@ -1448,7 +1448,10 @@ ShouldShutdownConnection(MultiConnection *connection, const int cachedConnection
 bool
 IsCitusInitiatedRemoteBackend(void)
 {
-	return application_name && strcmp(application_name, CITUS_APPLICATION_NAME) == 0;
+	return application_name &&
+		   strncmp(application_name, CITUS_APPLICATION_NAME_PREFIX,
+				   strlen(CITUS_APPLICATION_NAME_PREFIX)) == 0 &&
```

Comment:
seems hard to read. Can you simplify a bit? Maybe assign `strlen(CITUS_APPLICATION_NAME_PREFIX)` to a variable etc.

Also, `strisdigit_s(application_name + strlen(CITUS_APPLICATION_NAME_PREFIX), 20)` what is this? Where do we get 20?